File: src/dataset/cpp/food_chain_Exercism_thefullarcticfox/claude/ClaudeSonnet4_food_chain_v4.cpp
#include "food_chain.h"
#include <array>
#include <string_view>

namespace food_chain {
    struct AnimalData {
        std::string_view animal;
        std::string_view phrase;
    };
    
    static constexpr std::array<AnimalData, 8> animals = {{
        {"fly", "I don't know why she swallowed the fly. Perhaps she'll die."},
        {"spider", "It wriggled and jiggled and tickled inside her."},
        {"bird", "How absurd to swallow a bird!"},
        {"cat", "Imagine that, to swallow a cat!"},
        {"dog", "What a hog, to swallow a dog!"},
        {"goat", "Just opened her throat and swallowed a goat!"},
        {"cow", "I don't know how she swallowed a cow!"},
        {"horse", "She's dead, of course!"}
    }};
    
    std::string verse(int verse_num) {
        const int idx = verse_num - 1;
        std::string res;
        res.reserve(256);
        
        res += "I know an old lady who swallowed a ";
        res += animals[idx].animal;
        res += ".\n";
        
        if (verse_num > 1) {
            res += animals[idx].phrase;
            res += "\n";
        }
        
        if (verse_num == 8)
            return res;
            
        for (int i = verse_num - 1; i > 0; --i) {
            res += "She swallowed the ";
            res += animals[i].animal;
            res += " to catch the ";
            res += animals[i - 1].animal;
            if (i == 2)
                res += " that wriggled and jiggled and tickled inside her";
            res += ".\n";
        }
        
        res += animals[0].phrase;
        res += "\n";
        return res;
    }

    std::string verses(int first, int last) {
        std::string res;
        res.reserve((last - first + 1) * 256);
        
        for (int i = first; i <= last; ++i) {
            res += verse(i);
            res += "\n";
        }
        return res;
    }

    std::string sing() {
        return verses(1, 8);
    }
}

File: src/dataset/typescript/food-chain_Exercism-typescript-shybyte/claude/ClaudeSonnet4_food-chain_v4.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const FOOD_CHAIN = [
    { animal: 'fly', sentence: "I don't know why she swallowed the fly. Perhaps she'll die.", optionalDescription: '' },
    { animal: 'spider', sentence: 'It wriggled and jiggled and tickled inside her.', optionalDescription: ' that wriggled and jiggled and tickled inside her' },
    { animal: 'bird', sentence: 'How absurd to swallow a bird!', optionalDescription: '' },
    { animal: 'cat', sentence: 'Imagine that, to swallow a cat!', optionalDescription: '' },
    { animal: 'dog', sentence: 'What a hog, to swallow a dog!', optionalDescription: '' },
    { animal: 'goat', sentence: 'Just opened her throat and swallowed a goat!', optionalDescription: '' },
    { animal: 'cow', sentence: "I don't know how she swallowed a cow!", optionalDescription: '' },
    { animal: 'horse', sentence: "She's dead, of course!", optionalDescription: '' }
];
const food = (animal, sentence, optionalDescription = '') => ({
    animal, sentence, optionalDescription
});
function firstLineInVerse(animal) {
    return `I know an old lady who swallowed a ${animal}.`;
}
function verse(verseNumber) {
    if (verseNumber === 8) {
        const { animal, sentence } = FOOD_CHAIN[7];
        return `${firstLineInVerse(animal)}\n${sentence}\n`;
    }
    const lines = [];
    const foodChainPart = FOOD_CHAIN.slice(0, verseNumber);
    lines.push(firstLineInVerse(foodChainPart[verseNumber - 1].animal));
    lines.push(foodChainPart[verseNumber - 1].sentence);
    for (let i = verseNumber - 2; i >= 0; i--) {
        const current = foodChainPart[i];
        const previous = foodChainPart[i + 1];
        lines.push(`She swallowed the ${previous.animal} to catch the ${current.animal}${current.optionalDescription}.`);
    }
    if (verseNumber > 1) {
        lines.push(FOOD_CHAIN[0].sentence);
    }
    return lines.join('\n') + '\n';
}
function verses(minVerse = 1, maxVerse = 8) {
    let result = '';
    for (let i = minVerse; i <= maxVerse; i++) {
        result += verse(i);
        if (i < maxVerse)
            result += '\n';
    }
    return result;
}
exports.default = { verse, verses };
